funcs.go

```diff
@@ -0,0 +1,24 @@
+package microerror
+
+import (
+	"fmt"
+
+	"github.com/juju/errgo"
+)
+
+// Stack prints the error with the stack if its argument is underlying
+// microerror error or result of Error function otherwise. Its main purpose is
+// to be used for a value for "stack" micrologger key.
+//
+// Example:
+//
+//	logger.LogCtx(ctx, "level", "error", "message", "failed to do a thing", "stack", microerror.Stack(err))
+//
+func Stack(err error) string {
+	switch err.(type) {
+	case *errgo.Err:
+		return fmt.Sprintf("%#v", err)
+	default:
+		return err.Error()
```

Comment:
To understand this better: Are we expecting the specialized error types (e.g. the referenced `awsError`) to produce reasonable error message by invoking `Error()`? i.e. we don't loose that additional information that was present in those fields?

Reply:
I doubt any sane (and popular) library would expect user to print errors in a different way.
So yes I expect that and I feel pretty confident, but LMK if this is concerning and if you have an idea how to approach it.

Reply:
I agree with you that in general libraries should honor end users to call `Error()` and rely on that. I fear there are poor implementations, but I'm curious. Let's :shipit: